### PR TITLE
chore: use new ci-environment package instead of create-meta

### LINKED ISSUE #1867: Replace create-meta with ci-environment
The create-meta module has been [renamed to ci-environment](https://github.com/cucumber/ci-environment/pull/34). The create-meta module is no longer maintained.

### PR DESCRIPTION
**Pull request by davidjgoss**
# Motivation & context

Fixes #1867.

## Type of change

- Refactoring/debt (improvement to code design or tooling without changing behaviour)

# Checklist:

- [x] I have read the [**CONTRIBUTING**](../CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [x] I have added tests to cover my changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added an entry to the "Unreleased" section of the [**CHANGELOG**](../CHANGELOG.md), linking to this pull request.

### CODE REVIEW

File: src/cli/helpers.ts [hunk 1]
```diff
@@ -119,8 +120,37 @@ export async function emitMetaMessage(
...
+    runtime: {
+      name: 'node.js',
+      version: process.versions.node,
+    },
+    ci: ciEnvironment
+      ? {
+          ...ciEnvironment,
+          git: {
+            ...ciEnvironment.git,
+            remote: ciEnvironment.git.remote ?? '',
```
Comment (davidjgoss) [inline src/cli/helpers.ts:146]: @aslakhellesoy these fields are mandatory in `messages.Meta` but optional in `CiEnvironment`. Could we also make them optional in messages?

Comment (aslakhellesoy) [reply src/cli/helpers.ts:146]: @davidjgoss I think we should make `Git.remote` and `Git.revision` mandatory in `ci-environment` (they already are in `messages`). These fields are the bare minimum for using the data and if either of them are absent we shouldn't create a `Git` object at all. Both `branch` and `tag` can be undefined.

Comment (aslakhellesoy) [reply src/cli/helpers.ts:146]: See https://github.com/cucumber/ci-environment/issues/36

**Comment by aslakhellesoy** [comment]
I've released ci-environment 8.0.0 - let's update to that version before merging this.

File: src/cli/helpers.ts (hunk 1 above)
Comment (davidjgoss) [reply src/cli/helpers.ts]: Updated with the new library - much simpler now!

File: src/cli/helpers.ts [hunk 2]
```diff
@@ -118,8 +119,27 @@ export async function emitMetaMessage(
...
+    },
+    os: {
+      name: os.platform(),
+      version: os.release(),
+    },
+    runtime: {
+      name: 'node.js',
+      version: process.versions.node,
+    },
+    ci: detectCiEnvironment(env),
```
Comment (aslakhellesoy) [inline src/cli/helpers.ts:139]: The types are 100% compatible - that's neat :-)

**Pull request merged**